src/global/ApolloLinkState/persona.ts

```diff
@@ -0,0 +1,86 @@
+import { Resolvers } from 'apollo-boost';
+
+import { GET_CARD, GET_PAGE, GET_PERSONA_STEP_PATH } from 'global/graphqls/Persona';
+
+import { Persona, PersonaCard, PersonaPage } from './namespace';
+
+const cardDefaults: PersonaCard = {
+  __typename: 'PersonaCard',
+  name: '',
+  description: '',
+  avatar: '',
+  background: '',
+};
+
+const pageDefaults: PersonaPage = {
+  __typename: 'PersonaPage',
+  background: '',
+  avatar: '',
+  content: '',
+};
+
+const personaDefaults: Persona = {
+  __typename: 'Persona',
+  personaStepPath: '',
+  card: cardDefaults,
+  page: pageDefaults,
+  personaUUIDs: [],
+};
+
+const personaResolvers: Resolvers = {
+  Mutation: {
+    changePersonaStepPath: (_, { personaStepPath }: Persona, { cache }) => {
+      const previousState = cache.readQuery({ query: GET_PERSONA_STEP_PATH });
+
+      const data = {
+        persona: {
+          ...previousState.persona,
+          personaStepPath,
```

Comment:
Do we need to store this information if we have it in url address?

Reply:
this is reworked, please review it once more time